### PR TITLE
Modified all the Behaivor Tree's to work with lifecycle branch

### PR DESCRIPTION
**Pull request by mhpanah**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | () |
| Primary OS tested on | (18.04) |
| Robotic platform tested on | (Gazebo and TB3) |

---

## Description of contribution in a few bullet points

---

- Modified all the BT's to work with lifecycle branch
- Deleted unused BT's
- Renamed parallel tree to navigate_w_replanning
- Fixed isLocalized condition node. isLocalized condition node was using lifecycle node but Blackboard passes regular node.

Now we have three main tree's.  The first one which is set to default is `navigate with re-planning`, the second one is `navigate with re-planning and recovery` and the last one is `auto localization with re-planning and recovery`.
 
TODO: #773

### CODE REVIEW

**Comment by crdelsey** [comment]
Can you at least add hand drawn versions of the trees? They are very difficult to review as XML files. It's no more than 10 minutes work. If you don't do it, all your reviewers have to do it.

File: nav2_tasks/include/nav2_tasks/is_localized_condition.hpp [hunk 1]
```diff
@@ -93,11 +88,18 @@ class IsLocalizedCondition : public BT::ConditionNode
...
+    robot_.reset();
+  }
+
+private:
   static const int cov_x_ = 0;
   static const int cov_y_ = 7;
   static const int cov_a_ = 35;
 
-  nav2_lifecycle::LifecycleNode::SharedPtr node_;
+  rclcpp::Node::SharedPtr node_;
```
Comment (crdelsey) [inline nav2_tasks/include/nav2_tasks/is_localized_condition.hpp]: Why is this a normal node again?

**Comment by mhpanah** [comment]
BT

File: nav2_tasks/include/nav2_tasks/is_localized_condition.hpp (hunk 1 above)
Comment (mhpanah) [reply nav2_tasks/include/nav2_tasks/is_localized_condition.hpp]: I updated the PR description.

**Pull request merged**